source/codegen/validate-linux-rt.py

```diff
@@ -0,0 +1,73 @@
+"""Script for validating if a set of changed files warrants updating the Linux RT Feed.
+
+Expects output of git diff --name-only <last-release-commit> <current-commit> 
+to be piped in so it can be read from sys.stdin
+"""
+
+import argparse
+import fnmatch
+import os
+import sys
+from typing import List
+
+from template_helpers import load_metadata
+
+
+def _check_core_files_affecting_rt(changed_files: List[str]) -> bool:
+    files_affecting_linux_rt = [
+        "source/config/*",
```

Comment:
It seems like this is a little too easy to break by adding a new source folder. Those will silently default to not requiring a linux RT update. But shouldn't the opposite be the default? Or should it return an error for unexpected folders?

Reply:
Or is there a common profile of a change that doesn't affect RT that we specifically care about? Like adding a non-RT driver? Or a bugfix release that happens to not touch any code on RT?

Is this feature more about notifying *why* RT changed in a release than *whether* it changed?

Reply:
So the thought on this feature is to let whoever's making the grpc-device release know if they should update the Linux RT Feed after making the release. Essentially, letting them know whether they should go update the version of grpc-device Linux RT is pulling in: https://github.com/ni/meta-nilrt/blob/nilrt/master/hardknott/recipes-ni/ni-grpc-device/ni-grpc-device_git.bb

So this script will run based on a diff between the commit the draft release would be made from and the last release and try to tell the developer whether they will need to update the Linux RT Feed based on that.

With that all being said, this approach might not be the best. Right now it's working on a whitelist of RT supported driver updates and this set of files updating. But maybe it should be a blacklist of non-RT supported driver changes. So if there's any change other than one to a driver that doesn't support RT, we flag it as "needing to update Linux RT".

Reply:
We talked about this offline and I updated the approach to account for files we don't know about being added. The new approach is trying to see if the set of changes are likely just touching non-RT drivers to determine if the Linux RT feed should be updated or not.